### PR TITLE
Updated tests and project to work  #4

### PR DESCRIPTION
**Pull request by OsirisTerje**
I've added the adapter as a package, and then it comes alive.  
I do see that the random test is not working.  Added a new with fixed naming, and that doesn't work either.  
Also see that in the Mixed category there are some tests that say Theory_SomeCasesFail that do pass, but that is probably expected, since this is a theory test?  Some comments would be nice here

### CODE REVIEW

File: solutions/vs2017/NUnit3CoreTestDemo/NUnit3CoreTestDemo.csproj [hunk 1]
```diff
@@ -27,8 +20,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
-    <PackageReference Include="NUnit" Version="3.7.1" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.5.0" />
+    <PackageReference Include="NUnit" Version="3.9.0" />
+    <PackageReference Include="NUnit3TestAdapter" Version="3.9.0" />
```
Comment (CharliePoole) [inline solutions/vs2017/NUnit3CoreTestDemo/NUnit3CoreTestDemo.csproj:25]: If we add the adapter as a package, doesn't that make it a bit harder to test new versions of the adapter? My strategy was to leave the package out so I could just add it in when testing a new build manually. If we add it, does that mean I will have to edit the csproj file in order to test a new version?

**Review by CharliePoole — COMMENTED** [review]
I'm OK with merging this but I'd like to understand the reason for needing a reference to the package, which isn't used in any other of the demo projects.

File: solutions/vs2017/NUnit3CoreTestDemo/NUnit3CoreTestDemo.csproj (hunk 1 above)
Comment (OsirisTerje) [reply solutions/vs2017/NUnit3CoreTestDemo/NUnit3CoreTestDemo.csproj:25]: Earlier we tested using the vsix adapter, that doesnt work for .net core, and we're more and more moving towards a nuget adapter approach.  Yes, we need to edit it to test a new version, but it is just a  copy/replace over the csproj files.   And, I think it is essential to test this with Visual Studio itself, not just the command line runner.

**Comment by CharliePoole** [comment]
I just realized this demo is still not being run in CI.

**Pull request merged**